### PR TITLE
Fix tests

### PR DESCRIPTION
**Pull request by matt-bernhardt**

### CODE REVIEW

File: .travis.yml
Comment (DiegoPino) [inline .travis.yml]: Cool! We could also run `phpcs`  using the same find output. https://github.com/squizlabs/PHP_CodeSniffer

Comment (matt-bernhardt) [reply .travis.yml]: I was wondering about PHPCS too. For the theme development I do, I usually run a syntax checker like this, and then also a code analysis tool for conformance with code standards. Hearing you think the same way, I'll add it to this PR.

**Comment by matt-bernhardt** [comment]
Okay, at this point I've got a question for @DiegoPino and anyone else who wants to chime in. I've got PHPCS running correctly now, but there's 145 violations being flagged in the code - largely in boilerplate files that are distributed with Silex or the sample skeleton app.

This, to me, indicates that I'm most likely using the wrong coding standards. Is that accurate? What sort of code standards should I be using if not for the standard PHPCS ones?

The violations we are getting can be seen at https://travis-ci.org/matt-bernhardt/publibdata_ui/jobs/202119188 :

```
PHP CODE SNIFFER VIOLATION SOURCE SUMMARY
----------------------------------------------------------------------
    SOURCE                                                       COUNT
----------------------------------------------------------------------
[x] Squiz.Arrays.ArrayDeclaration.KeyNotAligned                  17
[x] Squiz.Arrays.ArrayDeclaration.CloseBraceNotAligned           14
[x] PEAR.WhiteSpace.ScopeIndent.Incorrect                        14
[x] PEAR.Functions.FunctionCallSignature.ContentAfterOpenBracke  11
[x] PEAR.Functions.FunctionCallSignature.CloseBracketLine        11
[x] PEAR.Functions.FunctionCallSignature.Indent                  10
[ ] Squiz.Commenting.InlineComment.InvalidEndChar                9
[x] PEAR.Functions.FunctionCallSignature.MultipleArguments       8
[ ] Squiz.Commenting.InlineComment.NotCapital                    8
[x] Squiz.WhiteSpace.SuperfluousWhitespace.EndLine               7
[ ] PEAR.Commenting.FileComment.Missing                          7
[x] Squiz.Arrays.ArrayDeclaration.ValueNotAligned                4
[x] Squiz.Commenting.PostStatementComment.Found                  3
[ ] Squiz.Operators.ComparisonOperatorUsage.ImplicitTrue         3
[x] PEAR.Functions.FunctionCallSignature.SpaceAfterCloseBracket  2
[x] Squiz.Arrays.ArrayDeclaration.MultiLineNotAllowed            2
[x] Squiz.Commenting.InlineComment.SpacingAfter                  2
[x] PEAR.Files.IncludingFile.UseInclude                          2
[x] Squiz.WhiteSpace.FunctionClosingBraceSpace.SpacingBeforeClo  1
[x] PEAR.Functions.FunctionDeclaration.SpaceAfterFunction        1
[x] Generic.Formatting.MultipleStatementAlignment.NotSame        1
[x] Squiz.Arrays.ArrayDeclaration.SingleLineNotAllowed           1
[x] Squiz.Commenting.ClosingDeclarationComment.Missing           1
[x] PEAR.Classes.ClassDeclaration.OpenBraceNewLine               1
[x] Squiz.Commenting.InlineComment.NoSpaceBefore                 1
[x] Generic.PHP.LowerCaseConstant.Found                          1
[ ] Squiz.Operators.ComparisonOperatorUsage.NotAllowed           1
[ ] PEAR.Commenting.FileComment.WrongStyle                       1
[ ] PEAR.Commenting.ClassComment.Missing                         1
----------------------------------------------------------------------
A TOTAL OF 145 SNIFF VIOLATIONS WERE FOUND IN 29 SOURCES
```

**Comment by DiegoPino** [comment]
@matt-bernhardt since we have little control to what others(composer required libraries) are contributing and how they are following simple `phpcs` coding standards, which also gets more complicated when certain libraries require fixed versions for their dependencies which could not have the latest fixes.

i would suggest we run `phpcs` only against our `/src` `/web` `/config` and `/test` folders? (i could have forgotten one..)

And also check if we are running latest PHP_CodeSniffer since some errors like Squiz.Arrays.ArrayDeclaration.KeyNotAligned  should have been fixed by
https://github.com/squizlabs/PHP_CodeSniffer/issues/433

What do you think?

**Comment by matt-bernhardt** [comment]
Oh, I definitely agree that we shouldn't be testing third-party code in these steps. To me, this means excluding the `/vendor` directory (and on Travis, exclusing the `php-codesniffer` directory itself). I've done this already, via exclusions in `codesniffer.ruleset.xml` and scoping the `find` statement in `.travis.yml` appropriately.

The question I raise above, however, has to do with violations found not in third-party code but in the actual application code:

```
Changing into directory /home/travis/build/matt-bernhardt/publibdata_ui/config
Processing dev.php [PHP => 99 tokens in 18 lines]... DONE in 8ms (15 errors, 0 warnings)
Processing prod.php [PHP => 82 tokens in 12 lines]... DONE in 6ms (9 errors, 0 warnings)
Changing into directory /home/travis/build/matt-bernhardt/publibdata_ui/src
Processing app.php [PHP => 227 tokens in 32 lines]... DONE in 16ms (19 errors, 0 warnings)
Processing console.php [PHP => 191 tokens in 23 lines]... DONE in 13ms (12 errors, 0 warnings)
Processing controllers.php [PHP => 620 tokens in 80 lines]... DONE in 43ms (79 errors, 0 warnings)
Changing into directory /home/travis/build/matt-bernhardt/publibdata_ui/src/controllers
Processing library.php [PHP => 40 tokens in 9 lines]... DONE in 3ms (4 errors, 0 warnings)
Changing into directory /home/travis/build/matt-bernhardt/publibdata_ui/web
Processing index.php [PHP => 60 tokens in 12 lines]... DONE in 4ms (2 errors, 0 warnings)
Processing index_dev.php [PHP => 139 tokens in 22 lines]... DONE in 9ms (5 errors, 0 warnings)
```

These files, in `/config`, `/src` and `/web` for now, _are_ under our control - so we should make sure they conform to whatever standards we adopt. My question is whether I'm using the right standard in the first place?

**Comment by DiegoPino** [comment]
@matt-bernhardt true. The Silex/Symfony2 standard is fuzzy. In general, it follows PSR1 and PSR2 of the by default installed standards (`phpcs -i`), more info [here](http://symfony.com/doc/current/contributing/code/standards.html)
There are some gray areas, like the type of [comments](https://squizlabs.github.io/PHP_CodeSniffer/analysis/silexphp/Silex/), etc, that can vary.

I would say let's use one these 2 options:
- https://github.com/leaphub/phpcs-symfony2-standard/blob/master/README.md
- https://github.com/djoos/Symfony2-coding-standard

I tested the second one:

```SHELL
phpcs --standard=Symfony2 src/app.php 

FILE: /Users/dpino/Desktop/bibdata/code/publibdata_ui/src/app.php
----------------------------------------------------------------------
FOUND 8 ERRORS AND 1 WARNING AFFECTING 7 LINES
----------------------------------------------------------------------
 11 | ERROR   | [x] TRUE, FALSE and NULL must be lowercase; expected
    |         |     "true" but found "TRUE"
 17 | ERROR   | [x] Expected 1 space after FUNCTION keyword; 0 found
 18 | ERROR   | [x] Line indented incorrectly; expected at least 4
    |         |     spaces, found 2
 18 | ERROR   | [ ] Add a comma after each item in a multi-line array
 20 | ERROR   | [x] Line indented incorrectly; expected at least 4
    |         |     spaces, found 2
 21 | WARNING | [ ] Please trim any trailing whitespace
 22 | ERROR   | [x] Line indented incorrectly; expected at least 4
    |         |     spaces, found 2
 23 | ERROR   | [x] Line indented incorrectly; expected at least 4
    |         |     spaces, found 2
 23 | ERROR   | [ ] Missing blank line before return statement
----------------------------------------------------------------------
PHPCBF CAN FIX THE 6 MARKED SNIFF VIOLATIONS AUTOMATICALLY
----------------------------------------------------------------------

Time: 47ms; Memory: 4.75Mb
````

**Comment by matt-bernhardt** [comment]
Ok, thanks. I'll try adding the Symfony2 standards and see if that doesn't result in a shorter list across the project. It may be that, for the sake of moving on, I propose this PR with just Travis/syntax/PHPUnit (which are all passing already), and send a second PR in after a bit with the PHPCS step and needed changes. Stay tuned, and thanks for your input so far!

**Comment by matt-bernhardt** [comment]
@DiegoPino Can you take a look at this now? I'm getting a clean report (finally!) from PHPCS, and the unit tests pass on PHP 7, but the first attempt failed on three of four tests under PHP 5.6 (which is what we're using on AWS, right?).

**Comment by DiegoPino** [comment]
@matt-bernhardt looks pretty awesome, good Job! Will give those tests a look under PHP 5.6 but we are running `PHP 7.0.13` and would say, even when backward compatibility is cool, we should aim for 7+ for now. 
Late afternoon: Will have more info and maybe a pull against this pull if I find a way of fixing these tests(or knowing why they fail) Thanks a lot!

**Comment by matt-bernhardt** [comment]
Oh, I thought we were running 5.6 in production. If that's on PHP 7.x, then I think for now we can drop 5.6 in the testing setup and merge this. We can take on backwards-compatibility as a separate piece of work later, if needed.

**Review by DiegoPino — APPROVED** [review]
Looks good! Having this makes future development strong. Very very grateful for your work @matt-bernhardt. If you want me to merge, I can do it 👍

**Comment by matt-bernhardt** [comment]
The practice we use in my group is that, among collaborators at least, the person who sends in the PR is the one who merges (after getting an approval from another collaborator) - so in that vein I'll merge this now. If we want to adopt a different practice for the future, then I'm open to that.

**Comment by DiegoPino** [comment]
@matt-bernhardt cool. We do the other way here, but I'm really really flexible and more than happy to just review and you merge your own Pulls 😁So lets keep it doing that way.

**Pull request merged**
